test/ably/realtimeconnection_test.py

```diff
@@ -125,3 +125,13 @@ def on_state_change(change):
         assert state_change.reason == exception.value
         assert ably.connection.error_reason == exception.value
         await ably.close()
+
+    async def test_realtime_request_timeout_connect(self):
```

Comment:
We should also add tests for channel attach/detach/ping

Reply:
Tests for attach, detach, close and ping cannot be added because connect has a timeout. With the test, it will timeout at `connect()` before it gets to the other protocol messages. I did comment out the `connect()` timeout in the cod to test the attach, detach, close and ping timeout individually.